### PR TITLE
uart: logging: Add configurable log level to UART drivers

### PR DESCRIPTION
**Pull request by pdietl**
This change adds compile-time selection of log level for
the UART drivers.

Signed-off-by: Pete Dietl <petedietl@gmail.com>

### CODE REVIEW

**Comment by pdietl** [comment]
@erwango once of the actions was cancelled for some reason. Can you make it run again?

**Review by erwango — DISMISSED** [review]

**Comment by erwango** [comment]
> @erwango once of the actions was cancelled for some reason. Can you make it run again?

Github now infroms that all checks are passed, so I'm assuming this is fine.

**Review by dcpleung — DISMISSED** [review]

File: drivers/serial/uart_ite_it8xxx2.c
Comment (mbolivar-nordic) [inline drivers/serial/uart_ite_it8xxx2.c:20]: Is it correct to leave LOG_LEVEL_ERR in place like this?

Comment (pdietl) [reply drivers/serial/uart_ite_it8xxx2.c:20]: Nope. Good catch

**Comment by pdietl** [comment]
@erwango can I get a re-approve? 
@mbolivar-nordic I think I fixed the issue.

File: drivers/serial/uart_npcx.c
Comment (ChiHuaL) [inline drivers/serial/uart_npcx.c:22]: Should we remove the LOG_LEVEL_ERR here?

Comment (pdietl) [reply drivers/serial/uart_npcx.c:22]: Yes.

**Review by dcpleung — DISMISSED** [review]

**Review by ChiHuaL — DISMISSED** [review]
Thanks.

**Review by erwango — DISMISSED** [review]

File: drivers/serial/uart_hvc_xen.c
Comment (henrikbrixandersen) [inline drivers/serial/uart_hvc_xen.c:23]:
```suggestion
#include <logging/log.h>
LOG_MODULE_REGISTER(uart_hvc_xen, CONFIG_UART_LOG_LEVEL);
```

I strongly prefer using this variant of the `LOG_MODULE_REGISTER()` macro as it makes it much more clear, which log level is used by a given file.

File: drivers/serial/uart_ite_it8xxx2.c
Comment (henrikbrixandersen) [inline drivers/serial/uart_ite_it8xxx2.c:20]: Same here. You are actually reverting from the two-argument `LOG_MODULE_REGISTER()` to a one-argument version.

File: drivers/serial/uart_neorv32.c
Comment (henrikbrixandersen) [inline drivers/serial/uart_neorv32.c:17]:
```suggestion
#include <logging/log.h>
LOG_MODULE_REGISTER(uart_neorv32, CONFIG_UART_LOG_LEVEL);
```

Which log level is supposed to be used here?

File: drivers/serial/uart_npcx.c
Comment (henrikbrixandersen) [inline drivers/serial/uart_npcx.c:22]: Same.

File: drivers/serial/uart_nrfx_uarte.c
Comment (henrikbrixandersen) [inline drivers/serial/uart_nrfx_uarte.c:21]: Same.

File: drivers/serial/uart_stm32.c
Comment (henrikbrixandersen) [inline drivers/serial/uart_stm32.c:40]: Same.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

**Review by dcpleung — DISMISSED** [review]

**Review by henrikbrixandersen — DISMISSED** [review]

**Review by erwango — DISMISSED** [review]

**Comment by pdietl** [comment]
@erwango @henrikbrixandersen @dcpleung Can I get approvals again?

**Review by henrikbrixandersen — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Comment by pdietl** [comment]
Can you merge then @henrikbrixandersen ? :)

**Comment by henrikbrixandersen** [comment]
> Can you merge then @henrikbrixandersen ? :)

Sorry, no merge rights.

**Review by erwango — APPROVED** [review]

**Pull request merged**
